From fc43a2bf668e325aab427a8c0827cfc3b8a1e39f Mon Sep 17 00:00:00 2001
From: popcornmix <popcornmix@gmail.com>
Date: Fri, 29 Apr 2016 17:27:35 +0100
Subject: [PATCH] Revert "bcm2835: add fallback channel layouts if channel map
 API is not used"

This reverts commit ceacfff9f86f89826dbc8a6df667f485894327d1.
---
 sound/arm/bcm2835-vchiq.c | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

--- a/sound/arm/bcm2835-vchiq.c
+++ b/sound/arm/bcm2835-vchiq.c
@@ -598,16 +598,7 @@ int bcm2835_audio_set_params(bcm2835_als
 	if (alsa_stream->chip->cea_chmap >= 0) {
 		chmap_value = (unsigned)alsa_stream->chip->cea_chmap << 24;
 	} else {
-		/* fallback layouts for applications which do not use chmap API */
-		chmap_value = 0x00;
-		switch (channels) {
-		case 3: chmap_value = 0x01; break;
-		case 4: chmap_value = 0x03; break;
-		case 5: chmap_value = 0x07; break;
-		case 6: chmap_value = 0x0b; break;
-		case 7: chmap_value = 0x0f; break;
-		case 8: chmap_value = 0x13; break;
-		}
+		chmap_value = 0; /* force stereo */
 		for (i = 0; i < 8; i++)
 			alsa_stream->chip->map_channels[i] = i;
 	}
